src/internal/passkey.ts

```diff
@@ -0,0 +1,174 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+/**
+ * This file contains the underlying implementations for exposed API surface in
+ * the {@link api/passkey}. By moving the methods out into a separate file,
+ * other namespaces and processes can access these methods without depending on the entire
+ * faucet namespace and without having a dependency cycle error.
+ */
+import { bufferToBase64URLString, startAuthentication, startRegistration } from "@simplewebauthn/browser";
+import {
+  VerifiedAuthenticationResponse,
+  VerifiedRegistrationResponse,
+  generateAuthenticationOptions as _generateAuthenticationOptions,
+  generateRegistrationOptions as _generateRegistrationOptions,
+  verifyAuthenticationResponse as _verifyAuthenticationResponse,
+  verifyRegistrationResponse as _verifyRegistrationResponse,
+} from "@simplewebauthn/server";
+import {
+  AuthenticationResponseJSON,
+  AuthenticatorDevice,
+  PublicKeyCredentialCreationOptionsJSON,
+  PublicKeyCredentialRequestOptionsJSON,
+  RegistrationResponseJSON,
+} from "@simplewebauthn/server/esm/deps";
+import { convertCOSEtoPKCS, cose, isoBase64URL, parseAuthenticatorData } from "@simplewebauthn/server/helpers";
+import { AptosConfig } from "../api/aptosConfig";
+import { AccountAddress, AccountPublicKey, AuthenticationKey, PublicKey } from "../core";
+import { PasskeyPublicKey } from "../core/crypto/passkey";
+import { Secp256r1PublicKey } from "../core/crypto/secp256r1";
+import { AccountAuthenticator, AnyRawTransaction, signWithPasskey as _signWithPasskey } from "../transactions";
+import { HexInput, PendingTransactionResponse } from "../types";
+import { AllowCredentialOption } from "../types/passkey";
+import { submitTransaction } from "./transactionSubmission";
+
+const supportedAlgorithmIDs = [cose.COSEALG.ES256];
+
+export async function generateRegistrationOptions(args: {
+  rpName: string;
+  rpID: string;
+  userID?: Uint8Array;
+  userName: string;
+  challenge?: string | Uint8Array;
+  userDisplayName?: string;
+  timeout?: number;
+  attestationType?: AttestationConveyancePreference;
+  authenticatorAttachment?: AuthenticatorAttachment;
+}): Promise<PublicKeyCredentialCreationOptionsJSON> {
+  const { authenticatorAttachment } = args;
+  const authenticatorSelection: AuthenticatorSelectionCriteria = {
+    authenticatorAttachment,
+    residentKey: "required",
+    userVerification: "required",
+  };
+
+  return _generateRegistrationOptions({
+    ...args,
+    authenticatorSelection,
+    supportedAlgorithmIDs,
+  });
+}
+
+export async function registerCredential(
```

Comment:
In addition to the registerCredential, there needs to be more methods verifying whether or not the credential was actually backedup (parse the registration response, verify backup eligibility and backup state), otherwise people will shoot themselves in the foot and create non-backedup credentials. 

See https://github.com/aptos-foundation/AIPs/blob/main/aips/aip-66.md#authenticator-data

Reply:
Additionally we need support for identifying incompetent / malicious authenticators via AAGUID.  see https://github.com/aptos-foundation/AIPs/blob/main/aips/aip-66.md#incompetent--malicious-authenticators

Reply:
1. Can I just expose this method `verifyAuthenticationResponse` to user? Its return type `VerifiedAuthenticationResponse` include `credentialDeviceType` and `credentialBackedUp`

2. Should we implement a policy that only allows the use of password managers on the list https://github.com/passkeydeveloper/passkey-authenticator-aaguids/blob/main/aaguid.json ?

Reply:
For context on my general thought process here, overall I'd like to provide strong / safe defaults on passkey registration for the user with the ability to override them if a developer wants to.

1. Yes you're on the right track, the `verifyAuthenticationResponse` method [parses `BE` and `BS`](https://github.com/MasterKale/SimpleWebAuthn/blob/master/packages/server/src/authentication/verifyAuthenticationResponse.ts#L237C56-L237C72) and returns the `VerifiedAuthenticationResponse`. To make it easier for the user we can just check to make sure both are set to true to determine if the credential is backed up.
2. Per the point above, yes, I think it may make sense to have a strong default for users that allows for authenticators that are well known and ~guaranteed~ to be backed up. `iCloud Keychain (Managed)`, `iCloud Keychain`, and `Google Password Manager` for example. `1Password` and `Dashlane` may also be acceptable but I'm not confident on the rest of them at this moment.